pkg/controller/dnsendpoint/dnsendpoint_controller.go

```diff
@@ -221,18 +270,17 @@ func (r *ReconcileDNSEndpoint) Reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
-func createNameServerQuery(c client.Client, logger log.FieldLogger) nameserver.Query {
-	awsCredsSecretName := os.Getenv(constants.ExternalDNSAWSCredsEnvVar)
-	if awsCredsSecretName != "" {
-		logger.Infof("using aws creds for external DNS stored in %q secret", awsCredsSecretName)
-		return nameserver.NewAWSQuery(c, awsCredsSecretName)
+func createNameServerQuery(c client.Client, logger log.FieldLogger, managedDomain hivev1.ManageDNSConfig) nameserver.Query {
+	if managedDomain.AWS != nil {
+		secretName := managedDomain.AWS.CredentialsSecretRef.Name
+		logger.Infof("using aws creds for managed domains stored in %q secret", secretName)
+		return nameserver.NewAWSQuery(c, secretName)
 	}
-
-	gcpCredsSecretName := os.Getenv(constants.ExternalDNSGCPCredsEnvVar)
-	if gcpCredsSecretName != "" {
-		logger.Infof("using gcp creds for external DNS stored in %q secret", gcpCredsSecretName)
-		return nameserver.NewGCPQuery(c, gcpCredsSecretName)
+	if managedDomain.GCP != nil {
+		secretName := managedDomain.GCP.CredentialsSecretRef.Name
+		logger.Infof("using gcp creds for managed domain stored in %q secret", secretName)
+		return nameserver.NewGCPQuery(c, secretName)
 	}
-
+	logger.Warn("unsupported cloud for managing DNS")
```

Comment:
This should be an error-level log entry. It is a logical error in the Hive code.